runtime/oti/j9nonbuilder.h

```diff
@@ -2960,7 +2971,7 @@ typedef struct J9Class {
 #define J9INTERFACECLASS_METHODORDERING(clazz) ((U_32*)((clazz)->finalizeLinkOffset))
 #define J9INTERFACECLASS_SET_METHODORDERING(clazz, value) (clazz)->finalizeLinkOffset = (UDATA)(value)
 
-#define J9ARRAYCLASS_SET_STRIDE(clazz, strideLength) ((clazz)->flattenedClassCache) = (J9FlattenedClassCache*)(UDATA)(strideLength)
+#define J9ARRAYCLASS_SET_STRIDE(clazz, strideLength) ((clazz)->flattenedClassCache) = (union J9FlattenedClassCache *)(UDATA)(strideLength)
```

Comment:
why is this change needed

Reply:
It seems to me that the compiler does not recognize J9FlattenedClassCache by itself. Without the union, the compiler will complain about there's no type name J9FlattenedClassCache.

Reply:
From my understanding, the reason why we can use J9FlattenedClassCache as a type 
without writing "struct J9FlattenedClassCache" every time was:
typedef struct J9FlattenedClassCache {
        .
        .
        .
} **J9FlattenedClassCache**;
Here, J9FlattenedClassCache can be used as a type. 
I suspect that union does not support this. I tried to do the following:
union J9FlattenedClassCache {
        struct J9FlattenedClassCacheEntry entry;
        struct J9FlattenedClassCacheHeader header;
} J9FlattenedClassCache;
This code caused the compiler to complain about multiple definitions of J9FlattenedClassCache.
However, I am not sure if it will work if I define the union structure and the type with two different names.

Reply:
I didn't know how to use typedef at that time. Fixed now.